src/js/App.js

```diff
@@ -1,70 +1,118 @@
-        />
+        {this.getStepIdx() !== BUTTON_STEP &&
+          this.getStepIdx() < this.state.steps.length - 1 && (
+            <Arrow
+              show={this.getStepIdx() < this.state.steps.length - 1}
+              direction="right"
+              handleOnClick={this.changeStep.bind(this, true)}
+            />
+          )}
+
+        {this.state.activeStep.key === "ideaBecomesABill" && (
+          <div className="buttons-container">
+            <Button
+              title="Take your bill to the Senate!"
+              handleOnClick={this.onButtonClick.bind(this, "SENATE")}
+              //handleOnClick={this.onButtonClick.bind(this, true)}
+            />
+            <Button
+              title="Take your bill to the House!"
+              handleOnClick={this.onButtonClick.bind(this, "HOUSE")}
+            />
+          </div>
+        )}
       </div>
-    )
+    );
   }
 
   changeStep = increase => {
-    const { activeStep } = this.state
-    const idx = STEPS.map(step => step.key).indexOf(activeStep.key)
-    const nextStep = increase ? STEPS[idx + 1] : STEPS[idx - 1]
+    const idx = this.getStepIdx();
+    const nextStep = increase
+      ? this.state.steps[idx + 1]
+      : this.state.steps[idx - 1];
     if (nextStep) {
-      this.setState({ activeStep: nextStep })
+      this.setState({ activeStep: nextStep });
     }
-  }
+  };
 
+  getStepIdx = () => {
+    console.log(this.state.steps);
+    console.log(this.state.activeStep);
+    return this.state.steps
+      .map(step => step.key)
+      .indexOf(this.state.activeStep.key);
+  };
   getStepClass = (step, idx) => {
-    const { activeStep } = this.state
+    const { activeStep } = this.state;
     if (step.key === activeStep.key) {
-      return "active-step"
+      return "active-step";
     }
-    const activeIdx = STEPS.map(step => step.key).indexOf(activeStep.key)
-    return idx > activeIdx ? "future-step" : "past-step"
-  }
+    const activeIdx = this.state.steps
+      .map(step => step.key)
+      .indexOf(activeStep.key);
+    return idx > activeIdx ? "future-step" : "past-step";
+  };
+
+  onButtonClick = nameOfButtonClicked => {
+    console.log("nameOfButtonClicked", nameOfButtonClicked);
+    console.log("Senate button clicked");
+    const earlySteps = STEPS.slice(0, 3);
+    console.log("earlySteps", earlySteps);
+    const houseSteps = STEPS.slice(3, 6);
+    console.log("houseSteps", houseSteps);
+    const senateSteps = STEPS.slice(6, 9);
+    console.log("senateSteps", senateSteps);
+
+    let stepProcess = [];
+
+    if (nameOfButtonClicked === "SENATE") {
+      stepProcess = [...earlySteps, ...senateSteps, ...houseSteps, STEPS[9]];
+      this.setState({ steps: stepProcess }, this.changeStep.bind(this, true));
+    } else if (nameOfButtonClicked === "HOUSE") {
+      stepProcess = [...earlySteps, ...houseSteps, ...senateSteps, STEPS[9]];
+      this.setState({ steps: stepProcess }, this.changeStep.bind(this, true));
+    }
+  };
```

Comment:
@madisoncarr nice work here. Clever!

Reply:
@wade00 Thanks! @pkrawc showed me how to do the ellipsis thing when making a new array.